### DISCUSSION TITLE
Just some tips, since I have a bit more expeirence making programing languages.

### DISCUSSION THREAD

**Discussion by MinekPo1** [Its-Jakey/SCPP-old-#7] [General]
1. Don't let the implementation influence the specification.

    Of course, its worth thinking about how you will implement the things you are adding to the spec, but if you are struggling to implement something, its often not worth to remove it from the spec.

2. Create an AST

    An AST or Abstract Syntax Tree, is a way of representing the code. Usually compilers look something like this:
    ```
                /-------\              /--------\           /-----------\
        code -> | Lexer | -> tokens -> | Parser | -> AST -> | Assembler | -> machine code
                \-------/              \--------/           \-----------/
    ```

    For reference your compiler looks something like this I believe:

    ```
                /-----------------------------\
                |           (antlr)           |
                | /-------\              /----------\
        code -> | | Lexer | -> tokens -> | "Parser" | -> machine code
                | \-------/              \----------/
                \-----------------------------/
    ```

3. Don't spam commits lol

    This is minor, but making tons of commits can look unprofessional haha

That's all the tips for now :)

**Comment by MinekPo1** [comment]
BTW you might want to read [Crafting Interpreters](https://craftinginterpreters.com/). Its a great resource and is available for free!

**Comment by MinekPo1** [comment]
Oh right I forgot, collect errors instead of using java exceptions.

Create a list, array, whatever and append errors to that. Then format the errors in a pretty way.

**Reply by Its-Jakey** [reply]
I started doing this in my new rewrite of the language

**Comment by Its-Jakey** [comment]
Thanks!
